Sources/BraintreeCore/ConfigurationCache.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+class ConfigurationCache {
+        
+    // MARK: - Private Properties
+    
+    private let timeToLiveMinutes = 5.0
+    
+    // MARK: - Internal Properties
+    
+    /// Exposed for testing
+    var cachedConfigStorage: [String: BTConfiguration] = [:]
```

Comment:
Note to self - see if NSCache functions synchronously. It offers auto eviction settings.

If not - implement max size on this dictionary.

Reply:
Moved from Dictionary to NSCache for the following reasons:
* Dictionary would require manual memory management. NSCache auto evicts if app memory consumption is high.
* NSCache is thread-safe. I was able to trigger a race condition using a very basic Dictionary setup. It would require more manual code to add locking.
* We can specify a max # of items, and max size in bytes of our NSCache, if we want